sql/plan/drop_trigger.go

```diff
@@ -0,0 +1,84 @@
+package plan
+
+import (
+	"fmt"
+	"github.com/dolthub/go-mysql-server/sql"
+	"strings"
+)
+
+type DropTrigger struct {
+	db          sql.Database
+	IfExists    bool
+	TriggerName string
+}
+
+var _ sql.Databaser = (*DropTrigger)(nil)
+var _ sql.Node = (*DropTrigger)(nil)
+
+// NewDropTrigger creates a new NewDropTrigger node for DROP TRIGGER statements.
+func NewDropTrigger(db sql.Database, trigger string, ifExists bool) *DropTrigger {
+	return &DropTrigger{
+		db:          db,
+		IfExists:    ifExists,
+		TriggerName: strings.ToLower(trigger),
+	}
+}
+
+// Resolved implements the sql.Node interface.
+func (d *DropTrigger) Resolved() bool {
+	_, ok := d.db.(sql.UnresolvedDatabase)
+	return !ok
+}
+
+// String implements the sql.Node interface.
+func (d *DropTrigger) String() string {
+	ifExists := ""
+	if d.IfExists {
+		ifExists = "IF EXISTS "
+	}
+	return fmt.Sprintf("DROP TRIGGER %s%s", ifExists, d.TriggerName)
+}
+
+// Schema implements the sql.Node interface.
+func (d *DropTrigger) Schema() sql.Schema {
+	return nil
+}
+
+// Children implements the sql.Node interface.
+func (d *DropTrigger) Children() []sql.Node {
+	return nil
+}
+
+// RowIter implements the sql.Node interface.
+func (d *DropTrigger) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	triggerDb, ok := d.db.(sql.TriggerDatabase)
+	if !ok {
+		if d.IfExists {
+			return sql.RowsToRowIter(), nil
+		} else {
+			return nil, sql.ErrTriggerDoesNotExist.New(d.TriggerName)
+		}
+	}
+	err := triggerDb.DropTrigger(ctx, d.TriggerName)
+	if err != nil && !sql.ErrTriggerDoesNotExist.Is(err) {
```

Comment:
Shouldn't be an error if the IfExists flag is set. Need a test for this

Reply:
Not sure what you mean here. If the `IF EXISTS` flag is set then it is _not_ an error. One of the tests does cover the `IF EXISTS` on a non-existent trigger as well.